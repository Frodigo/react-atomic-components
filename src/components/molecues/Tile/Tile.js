import React from 'react';
import PropTypes from 'prop-types';

import classes from './Tile.module.css';

const Tile = (props) => {
    const { heading, ...rest } = props;

    return (
        <section className={classes.tile} {...rest}>
            <h2 className={classes.heading}>{heading}</h2>
        </section>
    );
};

Tile.propTypes = {
    heading: PropTypes.string.isRequired
};

export default Tile;
