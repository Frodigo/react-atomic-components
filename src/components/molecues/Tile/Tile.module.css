.tile {
    align-items: center;
    background-color: var(--color-alternate);
    display: flex;
    justify-content: center;
    padding: 30px 15px;
}


.heading {
    color: var(--color-white);
    margin: 0;
}

@media (min-width: 768px) {
    .tile {
        padding: 60px 15px;
    }
}