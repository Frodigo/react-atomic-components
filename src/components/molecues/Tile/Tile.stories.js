import React from 'react';

import { default as TileComponent } from './Tile';

export default {
    title: 'Molecues/Tile',
    component: TileComponent
};

const Template = (args) => <TileComponent {...args} />;

export const Tile = Template.bind({});
Tile.args = {
    heading: 'This is Tile!'
};
