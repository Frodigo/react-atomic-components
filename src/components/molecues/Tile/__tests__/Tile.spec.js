import React from 'react';
import { render } from '@testing-library/react';

import Tile from '../Tile';

describe('Tile should be rendered', () => {
    const headingMock = 'This is tile';

    test('Tile renders correctly', () => {
        const { container } = render(<Tile heading={headingMock} />);

        expect(container).toMatchSnapshot();
    });
});
