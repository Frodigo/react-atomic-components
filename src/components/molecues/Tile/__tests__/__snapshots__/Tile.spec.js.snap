// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should be rendered Tile renders correctly 1`] = `
<div>
  <section
    class="tile"
  >
    <h2
      class="heading"
    >
      This is tile
    </h2>
  </section>
</div>
`;
